app/core/src/main/java/com/fsck/k9/mailstore/FolderSettingsProvider.kt

```diff
@@ -37,8 +38,4 @@ class FolderSettingsProvider(val preferences: Preferences, val account: Account)
 
         editor.commit()
     }
-
-    private fun String?.toFolderClass(defaultValue: FolderClass): FolderClass {
```

Comment:
I liked the extension function better. Please revert the changes to this class and copy the extension function to `RetrieveFolderOperations` (I prefer the code duplication over a global extension function on `String?`).

Reply:
Can you please tell me why you prefer code duplication over something like this:

```
package com.fsck.k9.mail

fun String?.toFolderClass(defaultValue: FolderClass): FolderClass {
    return if (this == null) defaultValue else FolderClass.valueOf(this)
}

enum class FolderClass {
    NO_CLASS, INHERITED, FIRST_CLASS, SECOND_CLASS
}
```
and `import com.fsck.k9.mail.toFolderClass` in `FolderSettingsProvider` and `RetrieveFolderOperations`?

Reply:
To limit the scope of the extension functions. `Storage` and `MailStore` are two independent stores that just happen to use the same serialization format for `FolderClass`. But this fact is an implementation detail that shouldn't be exposed.

Reply:
Done